ghostty package fixes

Jan200101 opened the pull request:
- uses correct source for ghostty 1.0.1 (1.0.0 used a generic `ghostty-source` filename)
- verify tarball using minisign
- set appropriate zig options when building ghostty:
    - DESTDIR is used to specifiy the build root because including it in the prefix may result in broken paths in the resulting output
    - `-Doptimize` has been superseded by `--release` in zig with both doing the same in the background
    - added extra flags to specifiy lib dir, exe dir and include dir to ensure zig picks the right stuff
- fetch zig packages ahead of time and enable system integration mode
    - I haven't verified if all the build requirements are correct but a local build shows that this results in proper dynamic linking

Jan200101 committed:
use correct source for ghostty 1.0.1

Jan200101 committed:
verify tarball for ghostty

Jan200101 committed:
set appropriate zig options when building ghostty

a breakdown:
- DESTDIR was added because the prefix is different from the build root, including it in the given prefix may include broken paths in the resulting output
- `-Doptimize` is the old version of `--release` and is around for backwards compatibility
- extra flags for lib dir, exe dir and include dir were added to ensure zig install stuff to the right location

Jan200101 committed:
fetch zig packages ahead of time and enable system integration

GildedRoach commented:
Thank you for these, I hadn't gotten to fixing them yet.

Though for the cache one, is this needed when doing online builds?

Jan200101 commented:
> Though for the cache one, is this needed when doing online builds?

its not needed *however* enabling system integration mode makes it link against system libraries for most dependencies so its definitely a benefit.

the only system integration it doesn't enable are:
- simdutf
- glslang 
- spirv-cross

simdutf can be explicitly enabled for F41+ but its not really needed, glslang is too new in fedora and spirv-cross has not been packaged

GildedRoach commented:
> > Though for the cache one, is this needed when doing online builds?
> 
> its not needed _however_ enabling system integration mode makes it link against system libraries for most dependencies so its definitely a benefit.
> 
> the only system integration it doesn't enable are:
> 
>     * simdutf
> 
>     * glslang
> 
>     * spirv-cross
> 
> 
> simdutf can be explicitly enabled for F41+ but its not really needed, glslang is too new in fedora and spirv-cross has not been packaged

Thank you for your explanation! This helps. I was not aware the cache was important to dynamic linking in any way, as it was explained in the packaging docs I understood it to be for offline building.

Jan200101 commented:
to be clear, the cache is not needed for this, you can also just explicitly set `-fsys=<dep>` for every single dependency and it works just fine, using `--system` simply enables most by default

GildedRoach commented:
> to be clear, the cache is not needed for this, you can also just explicitly set `-fsys=<dep>` for every single dependency and it works just fine, using `--system` simply enables most by default

Thank you. I probably would have done the `-fsys` method so the build has less to download, but if this is also acceptable to those with merging power, that works too.

Jan200101 committed:
add build requirements for dynamic linking

GildedRoach committed:
Update ghostty.spec

Signed-off-by: Gil <rockgrub@protonmail.com>

GildedRoach committed:
Update ghostty-nightly.spec

Signed-off-by: Gil <rockgrub@protonmail.com>

Jan200101 commented:
- reordered changelog to make various tooling happy
- added build requirements for dynamic linking to work, verified using mock
- updated license to match everything vendored in ghostty or included in the final binary:
```
# ghostty:                  MIT
# libvaxis:                 MIT
# libxev:                   MIT
# zig-objc:                 MIT
# zig-js:                   MIT
# z2d:                      MPL-2.0
# zf:                       MIT
# zigimg:                   MIT
# ziglyph:                  MIT
# zg:                       MIT
# iTerm2-Color-Schemes:     MIT
# CodeNewRoman              OFL-1.1
# GeistMono                 OFL-1.1
# Inconsolata               OFL-1.1
# JetBrainsMono             OFL-1.1
# JuliaMono                 OFL-1.1
# KawkabMono                OFL-1.1
# Lilex                     OFL-1.1
# MonaspaceNeon             OFL-1.1
# NotoEmoji                 OFL-1.1
# CozetteVector             MIT
# NerdFont                  MIT AND OFL-1.1
```

Jan200101 committed:
Update license to match everything vendored

Jan200101 committed:
reorder changelog to be in chronological order

raboneko commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|f40|Backport failed because of merge conflicts|
|✅|f41|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/2835">](https://github.com/terrapkg/packages/pull/2835)|
|❌|el10|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
backport --pr 2834
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/12571432370) for details

<!--- Backport version: 9.3.0 -->
